public/app/plugins/panel/stat/types.ts

```diff
@@ -65,6 +73,34 @@ export function addStandardDataReduceOptions(
     showIf: currentConfig => currentConfig.reduceOptions.values === false,
   });
 
+  if (includeFieldMatcher) {
+    builder.addSelect({
+      path: 'reduceOptions.fields',
+      name: 'Fields',
+      description: 'Select the fields that should be included in the panel',
+      settings: {
+        allowCustomValue: true,
+        options: [],
+        getOptions: (context: FieldOverrideContext) => {
```

Comment:
This is another reason for me to introduce panel options context. We are using "field override" semantics in the context of panel options which is very confusing.

Reply:
BTW - great that you have added the `getOptions` option 👍  Think longer term we will need it to be async as well.